src/Illuminate/Foundation/Application.php

```diff
@@ -1120,7 +1127,11 @@ public function getCachedConfigPath()
      */
     public function routesAreCached()
     {
-        return $this['files']->exists($this->getCachedRoutesPath());
+        if (is_null($this->routesAreCached)) {
+            $this->routesAreCached = $this['files']->exists($this->getCachedRoutesPath());
+        }
+
+        return $this->routesAreCached;
```

Comment:
```suggestion
        return $this->routesAreCached ??= $this['files']->exists($this->getCachedRoutesPath());
```

Kind of a nit-pick, but this does the same thing and is shorter (also by doing an isset check the explicit null default becomes unnecessary)